### ISSUE TITLE
Make accessing `time.self` a bit more ergonomic...

### ISSUE THREAD

**Issue by rwjblue** [heimdalljs/heimdalljs-graph#19]
As far as I can tell, the only (non-underscored) way to get a given nodes self time is something like:

```js
function selfTime(node) {
  for (let [statName, value] of node.statsIterator()) {
    if (statName === 'time.self') {
      return value;
    }
  }
}
```

Which is actually pretty annoying if all you need is the self time...

I do not have a specific API in mind, but I feel like we can come up with something better.

**Comment by rwjblue** [comment]
One suggestion would be something like (which basically just emulates the underlying data structure):

```js
node.stats.time.self
```

Another idea is something like:

```js
node.getStat('time.self');
```

Likely many other possibilities too...

**Comment by hjdivad** [comment]
Yes.  The motivation for the iterators is as primitives upon which something better can be built, but which also serve as an escape valve when the nicer API misses cases.  I think we may want to implement (more efficiently) things similar to what we did in the spike, with `sumStat`, `groupBy` &c.

Stef had a spike at one point that was chainable which is likely nice for building up a plan and then executing it efficiently (so we don't have to iterate the graph repeatedly).
